spec/lib/TemplateSrvStub.ts

```diff
@@ -1,26 +1,63 @@
-import { ScopedVars, TimeRange } from '@grafana/data';
+import { ScopedVars, TimeRange, TypedVariableModel, AdHocVariableFilter } from '@grafana/data';
 import { TemplateSrv } from '@grafana/runtime/services/templateSrv';
-import template from 'lodash/template';
 
 export default class TemplateSrvStub implements TemplateSrv {
   containsTemplate(target?: string | undefined): boolean {
     throw new Error('Method not implemented.');
   }
+
   updateTimeRange(timeRange: TimeRange): void {
     throw new Error('Method not implemented.');
   }
+
   templateSettings = { interpolate: /\[\[([\s\S]+?)\]\]/g };
   data = {};
+  variables: TypedVariableModel[] | [] = [];
   adhocFilters = [];
   // tslint:disable-next-line:max-line-length
-  // https://github.com/grafana/grafana/blob/c1c0daa13d5951e3e13d46ef9467d9d832993fe5/public/app/features/variables/utils.ts#L23
+  // https://github.com/grafana/grafana/blob/ab5a3820d5fff4f0f042024b0aee0632e6a4ca08/public/app/features/variables/utils.ts#L24
+  /*
+   * This regex matches 3 types of variable reference with an optional format specifier
+   * There are 6 capture groups that replace will return
+   * \$(\w+)                                    $var1
+   * \[\[(\w+?)(?::(\w+))?\]\]                  [[var2]] or [[var2:fmt2]]
+   * \${(\w+)(?:\.([^:^\}]+))?(?::([^\}]+))?}   ${var3} or ${var3.fieldPath} or ${var3:fmt3} (or ${var3.fieldPath:fmt3} but that is not a separate capture group)
+   */
   regex = /\$(\w+)|\[\[(\w+?)(?::(\w+))?\]\]|\${(\w+)(?:\.([^:^\}]+))?(?::([^\}]+))?}/g;
-  replace(target?: string, scopedVars?: ScopedVars, format?: string | Function): string {
-    return template(target, this.templateSettings)(this.data);
+
+  // from: https://github.com/grafana/grafana/blob/ab5a3820d5fff4f0f042024b0aee0632e6a4ca08/public/app/features/templating/template_srv.mock.ts#L30
+  // Adjusted for use of scoped variables, and ad hoc filters.
+  replace(target?: string, scopedVars?: ScopedVars): string {
+    if (!target) {
```

Comment:
use strict comparisons pls 

```suggestion
    if (target === undefined) {
```